mvc/controllers/actions.rst

```diff
@@ -1,8 +1,77 @@
+--------------------
+
+In ASP.NET MVC, a `Controller` is used to define and group a set of actions. An `action` (or `action method`) is a method on a controller that handles incoming requests. Controllers provide a logical means of grouping similar actions together, allowing common sets of rules (e.g. routing, caching, authorization) to be applied collectively. Incoming requests are mapped to actions through :doc:`routing <routing>`.
+
+In ASP.NET 5, a controller can be any instantiable class that ends in "Controller" or inherits from a class that ends with "Controller". Controllers should follow the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle>`_ and request any dependencies their actions require through their constructor using :doc:`dependency injection <dependency-injection>`.
+
+By convention, controller classes:
+	* are located in the root-level "Controllers" folder
+	* inhert from Microsoft.AspNet.Mvc.Controller
+
+These two conventions are not required.
+
+Within the Model-View-Controller pattern, a Controller is responsible for the initial processing of the request and instantiation of the Model. Generally, business decisions should  be performed within the Model.
+
+.. note:: The Model should be a `Plain Old CLR Object (POCO)`, not a ``DbContext`` or database-related type.
+
+The controller takes the result of the model's processing (if any), returns the proper view along with the associated view data. Learn more: :doc:`/overview` and :doc:`/getting-started/first-mvc-app/start-mvc`.
+
+.. tip:: The Controller is a `UI level` abstraction. Its responsibility is to ensure incoming request data is valid and to choose which view (or result for an API) should be returned. In well-factored apps it will not directly include data access or business logic, but instead will delegate to services handling these responsibilities.
+ 
+Defining Actions
+----------------
+Any public method on a controller type is an action. Parameters on actions are bound to request data and validated using :doc:`model binding </models/model-binding>`.
+
+.. warning:: Action methods that accept parameters should verify the ``ModelState.IsValid`` property is true.
+
+Action methods should contain logic for mapping an incoming request to a business concern. Business concerns should typically be represented as services that your controller accesses through :doc:`dependency injection <dependency-injection>`. Actions then map the result of the business action to an application state.
+
+Actions can return anything, but frequently will return an instance of ``IActionResult`` (or ``Task<IActionResult>`` for async methods) that produces a response. The action method is responsible for choosing `what kind of response`; the action result `does the responding`. You can either create and return your own ``ActionResult`` instance directly, or you can use one of the many helper methods available on the Controller base class.
```

Comment:
I think we should just remove any mention of `ActionResult` and focus on `IActionResult`. `ActionResult` is really only there to make porting code easier - most people should just completely ignore it. Technically there is a benefit of returning `IActionResult` - every action result is an `IActionResult`, but not every action result derives from `ActionResult`.